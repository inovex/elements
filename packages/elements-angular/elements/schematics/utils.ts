import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
