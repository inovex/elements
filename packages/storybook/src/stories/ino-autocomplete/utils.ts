import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
