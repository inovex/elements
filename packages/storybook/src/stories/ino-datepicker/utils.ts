import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
