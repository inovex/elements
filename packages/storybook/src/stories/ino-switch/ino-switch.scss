@use '../../../../elements/src/components/base/theme';

.story-switch {
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
}

.customizable-switch {
  display: block;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 40px;
  align-self: center;
}

ino-switch {
  margin: 5px;
}
