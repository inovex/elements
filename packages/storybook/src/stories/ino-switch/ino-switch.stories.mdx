import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-switch/readme.md";

<Meta title="Input/ino-switch/Documentation" />

<Description markdown={readme} />
