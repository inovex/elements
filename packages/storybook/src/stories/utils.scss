/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
