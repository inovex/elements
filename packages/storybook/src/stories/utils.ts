import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
