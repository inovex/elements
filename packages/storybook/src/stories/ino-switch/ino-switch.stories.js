import { boolean, select, text } from '@storybook/addon-knobs';

import withStencilReadme from '_local-storybookcore/with-stencil-readme';

import componentReadme from '_local-elements/src/components/ino-switch/readme.md';
import './ino-switch.scss';
import { INO_COLOR_SCHEMES } from '../../core/helpers/constants';

// https://github.com/storybooks/storybook/issues/4337#issuecomment-428495664
function subscribeToComponentEvents() {
  // == event block
  const eventHandler = function (e) {
    const el = e.target;

    if (el.tagName.toLowerCase() !== 'ino-switch' || el.id !== 'custom-switch') {
      return;
    }

    el.setAttribute('checked', e.detail);
  };

  document.addEventListener('checkedChange', eventHandler);
  // == event block

  // unsubscribe function will be called by Storybook
  return () => {
    document.removeEventListener('checkedChange', eventHandler);
  };
}

export default {
  title: 'Input/<ino-switch>',

  decorators: [
    withStencilReadme(componentReadme),
    (story) => {
      subscribeToComponentEvents();
      return story();
    },
  ],
};

export const DefaultUsage = () => /*html*/ `
<div>
<h4>Customizable ino-switch</h4>
<div>
  <style>
      ino-switch.customizable-switch {
        --switch-toggled-on-color: ${text('--switch-toggled-on-color', '#3d40f5', 'Custom Properties')};
        --switch-toggled-on-thumb-color: ${text('--switch-toggled-on-thumb-color', '#3d40f5', 'Custom Properties')};
        --switch-toggled-on-track-color: ${text('--switch-toggled-on-track-color', '#5d60f7', 'Custom Properties')};
        --switch-toggled-off-color: ${text('--switch-toggled-off-color', '#c1c1c1', 'Custom Properties')};
        --switch-toggled-off-thumb-color: ${text('--switch-toggled-off-thumb-color', '#fff', 'Custom Properties')};
        --switch-toggled-off-track-color: ${text('--switch-toggled-off-track-color', '#777777', 'Custom Properties')};
        --switch-hover-color: ${text('--switch-hover-color', '#5d60f7', 'Custom Properties')};
        --switch-hover-thumb-color: ${text('--switch-hover-thumb-color', '#fff', 'Custom Properties')};
        --switch-hover-track-color: ${text('--switch-hover-track-color', '#5d60f7', 'Custom Properties')};
        --switch-active-color: ${text('--switch-active-color', '#0d10f3', 'Custom Properties')};
        --switch-active-thumb-color: ${text('--switch-active-thumb-color', '#fff', 'Custom Properties')};
        --switch-active-track-color: ${text('--switch-active-track-color', '#0d10f3', 'Custom Properties')};
        --switch-disabled-color: ${text('--switch-disabled-color', '#c1c1c1', 'Custom Properties')};
        --switch-disabled-thumb-color: ${text('--switch-disabled-thumb-color', '#c1c1c1', 'Custom Properties')};
        --switch-disabled-track-color: ${text('--switch-disabled-track-color', '#c1c1c1', 'Custom Properties')};
      }
   </style>
   <ino-switch
     id="custom-switch"
     class="customizable-switch"
     checked="${boolean('checked', false)}"
     disabled="${boolean('disabled', false)}"
   >
      ${text('<slot />', 'Customizable Switch')}
   </ino-switch>
 </div>

 <h4>Color Schemes</h4>
 <div class="ino-switch-colors">
    ${INO_COLOR_SCHEMES.map(
      (color) => `<ino-switch ino-color-scheme="${color}" checked>${color}</ino-switch>`
    )
      .join()
      .replace(/,/g, '')}
</div>

<h4>States</h4>
 <div class="story-switch-states">
    <ino-switch checked>Checked</ino-switch>
    <ino-switch>Unchecked</ino-switch>
    <ino-switch disabled checked>Checked and Disabled</ino-switch>
    <ino-switch disabled>Disabled</ino-switch>
  </div>
</div>
`;
