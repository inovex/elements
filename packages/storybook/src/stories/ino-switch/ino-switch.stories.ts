import { useEffect } from '@storybook/client-api';
import { Components } from '@inovex.de/elements';
import { Story } from '@storybook/web-components';
import { html } from 'lit-html';
import { defaultDecorator, withColorScheme } from '../utils';
import './ino-switch.scss';

const eventHandler = (e) => {
  const el = e.target;
  if (el.tagName.toLowerCase() !== 'ino-switch') {
    return;
  }
  el.setAttribute('checked', e.detail);
};

export default {
  title: 'Input/ino-switch',
  component: 'ino-switch',
  decorators: [
    (story) => defaultDecorator(story, 'story-switch'),
    (story) => {
      useEffect(() => {
        document.addEventListener('checkedChange', eventHandler);
        return () =>
          document.removeEventListener('checkedChange', eventHandler);
      });
      return story();
    },
  ],
};

export const Playground: Story<Components.InoSwitch> = (args) => html`
  <ino-switch
    checked="${args.checked}"
    color-scheme="${args.colorScheme}"
    disabled="${args.disabled}"
    name="${args.name}"
  >
    Switch Label
  </ino-switch>
`;
Playground.args = {
  checked: false,
  disabled: false,
  name: '',
};
withColorScheme(Playground, 'colorScheme', 'primary');

export const ColorScheme = () => html`
  <ino-switch color-scheme="primary" checked>Primary</ino-switch>
  <ino-switch color-scheme="secondary" checked>Secondary</ino-switch>
  <ino-switch color-scheme="success" checked>Success</ino-switch>
  <ino-switch color-scheme="warning" checked>Warning</ino-switch>
  <ino-switch color-scheme="error" checked>Error</ino-switch>
  <ino-switch color-scheme="light" checked>Light</ino-switch>
  <ino-switch color-scheme="dark" checked>Dark</ino-switch>
`;

export const States = () => html`
  <ino-switch checked>Checked</ino-switch>
  <ino-switch>Unchecked</ino-switch>
  <ino-switch disabled checked>Checked and Disabled</ino-switch>
  <ino-switch disabled>Disabled</ino-switch>
`;

export const CSSProperties = (args) => html`
  <style>
    ino-switch.css-variables {
      --switch-toggled-on-color: ${args.switchToggledOnColor};
      --switch-toggled-on-thumb-color: ${args.switchToggledOnThumbColor};
      --switch-toggled-on-track-color: ${args.switchToggledOnTrackColor};
      --switch-toggled-off-color: ${args.switchToggledOffColor};
      --switch-toggled-off-thumb-color: ${args.switchToggledOfThumbColor};
      --switch-toggled-off-track-color: ${args.switchToggledOfTrackColor};
      --switch-hover-color: ${args.switchHoverColor};
      --switch-hover-thumb-color: ${args.switchHoverThumbColor};
      --switch-hover-track-color: ${args.switchHoverTrackColor};
      --switch-active-color: ${args.switchActiveColor};
      --switch-active-thumb-color: ${args.switchActiveThumbColor};
      --switch-active-track-color: ${args.switchActiveTrackColor};
      --switch-disabled-color: ${args.switchDisabledColor};
      --switch-disabled-thumb-color: ${args.switchDisabledThumbColor};
      --switch-disabled-track-color: ${args.switchDisabledTrackColor};
    }
  </style>
  <ino-switch
    class="css-variables"
    checked="${args.checked}"
    disabled="${args.disabled}"
  >
    Css Variables Switch
  </ino-switch>
`;
CSSProperties.args = {
  checked: false,
  disabled: false,
  switchToggledOnColor: '#3d40f5',
  switchToggledOnThumbColor: '#3d40f5',
  switchToggledOnTrackColor: '#5d60f7',
  switchToggledOffColor: '#c1c1c1',
  switchToggledOfThumbColor: '#fff',
  switchToggledOfTrackColor: '#777777',
  switchHoverColor: '#5d60f7',
  switchHoverThumbColor: '#fff',
  switchHoverTrackColor: '#5d60f7',
  switchActiveColor: '#0d10f3',
  switchActiveThumbColor: '#fff',
  switchActiveTrackColor: '#0d10f3',
  switchDisabledColor: '#c1c1c1',
  switchDisabledThumbColor: '#c1c1c1',
  switchDisabledTrackColor: '#c1c1c1',
};
CSSProperties.argTypes = {
  switchToggledOnColor: { control: 'color' },
  switchToggledOnThumbColor: { control: 'color' },
  switchToggledOnTrackColor: { control: 'color' },
  switchToggledOffColor: { control: 'color' },
  switchToggledOfThumbColor: { control: 'color' },
  switchToggledOfTrackColor: { control: 'color' },
  switchHoverColor: { control: 'color' },
  switchHoverThumbColor: { control: 'color' },
  switchHoverTrackColor: { control: 'color' },
  switchActiveColor: { control: 'color' },
  switchActiveThumbColor: { control: 'color' },
  switchActiveTrackColor: { control: 'color' },
  switchDisabledColor: { control: 'color' },
  switchDisabledThumbColor: { control: 'color' },
  switchDisabledTrackColor: { control: 'color' },
};
