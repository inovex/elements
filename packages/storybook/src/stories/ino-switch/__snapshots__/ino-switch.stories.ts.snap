// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input/ino-switch Checked smoke-test 1`] = `
<div id="root-inner">
  <div class="story-switch">
    <ino-switch checked
                color-scheme="primary"
                disabled="false"
                name
                class="ino-switch ino-switch--color-scheme-primary hydrated"
    >
      <button id="ino-switch-id__5ao13o3rl"
              class="mdc-switch mdc-switch--selected mdc-ripple-upgraded mdc-ripple-upgraded--unbounded"
              type="button"
              role="switch"
              aria-checked="true"
              style="--mdc-ripple-fg-size:0px; --mdc-ripple-fg-scale:NaN; --mdc-ripple-left:0px; --mdc-ripple-top:0px;"
      >
        <div class="mdc-switch__track">
        </div>
        <div class="mdc-switch__handle-track">
          <div class="mdc-switch__handle">
            <div class="mdc-switch__ripple">
            </div>
          </div>
        </div>
      </button>
      <label for="ino-switch-id__5ao13o3rl">
        Switch Label
        <input type="hidden"
               class="aux-input"
               name
               value
        >
      </label>
    </ino-switch>
  </div>
</div>
`;

exports[`Input/ino-switch Disabled smoke-test 1`] = `
<div id="root-inner">
  <div class="story-switch">
    <ino-switch checked="false"
                color-scheme="primary"
                disabled
                name
                class="ino-switch ino-switch--color-scheme-primary ino-switch-disabled hydrated"
    >
      <button id="ino-switch-id__10bm7wkda"
              class="mdc-switch mdc-switch--unselected mdc-ripple-upgraded mdc-ripple-upgraded--unbounded"
              disabled
              type="button"
              role="switch"
              aria-checked="false"
              style="--mdc-ripple-fg-size:0px; --mdc-ripple-fg-scale:NaN; --mdc-ripple-left:0px; --mdc-ripple-top:0px;"
      >
        <div class="mdc-switch__track">
        </div>
        <div class="mdc-switch__handle-track">
          <div class="mdc-switch__handle">
            <div class="mdc-switch__ripple">
            </div>
          </div>
        </div>
      </button>
      <label for="ino-switch-id__10bm7wkda">
        Switch Label
        <input type="hidden"
               class="aux-input"
               disabled
               name
               value
        >
      </label>
    </ino-switch>
  </div>
</div>
`;

exports[`Input/ino-switch Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-switch">
    <ino-switch checked="false"
                color-scheme="primary"
                disabled="false"
                name
                class="ino-switch ino-switch--color-scheme-primary hydrated"
    >
      <button id="ino-switch-id__ibqqacosg"
              class="mdc-switch mdc-switch--unselected mdc-ripple-upgraded mdc-ripple-upgraded--unbounded"
              type="button"
              role="switch"
              aria-checked="false"
              style="--mdc-ripple-fg-size:0px; --mdc-ripple-fg-scale:NaN; --mdc-ripple-left:0px; --mdc-ripple-top:0px;"
      >
        <div class="mdc-switch__track">
        </div>
        <div class="mdc-switch__handle-track">
          <div class="mdc-switch__handle">
            <div class="mdc-switch__ripple">
            </div>
          </div>
        </div>
      </button>
      <label for="ino-switch-id__ibqqacosg">
        Switch Label
        <input type="hidden"
               class="aux-input"
               name
               value
        >
      </label>
    </ino-switch>
  </div>
</div>
`;
